src/middlewared/middlewared/test/integration/utils/files.py

```diff
@@ -0,0 +1,40 @@
+import sys
+import pathlib
+import contextlib
+
+try:
+    apifolder = pathlib.Path().cwd().as_posix()
+    sys.path.append(apifolder)
+    from auto_config import user, password, ip
+    from functions import get_file
+except ImportError:
+    pass
+
+__all__ = ["get_remote_file_temporarily"]
+
+
+@contextlib.contextmanager
+def get_remote_file_temporarily(file_to_get, destination, _ip=None):
+    _ip = _ip if _ip is not None else ip
+    remote_file = pathlib.Path(file_to_get)
+    if not remote_file.is_absolute():
+        assert False, 'An absolute path to the file being downloaded must be provided'
+
+    local = pathlib.Path(destination)
+    if not local.is_absolute():
+        assert False, 'An absolute path for where the file is downloaded must be provided'
+    elif local.is_dir():
+        local_absolute = local.joinpath(remote_file).as_posix()
+    else:
+        local_absolute = local.as_posix()
+
+    gotit = None
+    try:
+        gotit = get_file(remote_file.as_posix(), local_absolute, user, password, _ip)
```

Comment:
We only use this for small text files that fit into memory, can we just run `cat /file/path` via SSH in order to avoid using temporary files in the filesystem? That will result in a much more easy-to-use API

Reply:
And what about files that are bigger?

Reply:
We can keep the method that will retrieve them into a temporary path, but a method to retrieve to RAM will be more convenient for the vast majority of the test cases. Also let's remember that this is a test runner so even 100MB file is fine to be read in RAM.